### PR TITLE
Remove judge name from case details hearing section

### PR DESCRIPTION
**Pull request by eileen-nava**
Resolves [#1757: Hide Judge Name in Hearings Section](https://vajira.max.gov/browse/CASEFLOW-1757).

### Description
We don't want VSOs/POAs to be able to see the judge who is assigned to a hearing, because we don't want them to have access to this information before a hearing.  I shielded this information in the Case Details Page's Hearings Section.  Please see attached screenshots. 

### Acceptance Criteria
- [ ] Code compiles correctly
- [ ] VSO user cannot see judge name on case details
- [ ] Non-vso user can see judge name on case details page

### Testing Plan
1. Log in as `BILLIE_VSO`

2. Create a veteran, appeal, and associated hearing in the rails console.
```
RequestStore[:current_user] = User.system_user
veteran = FactoryBot.create(:veteran, first_name: "Chibueze", last_name: "Vanscoy", file_number: 800_888_001)
appeal = FactoryBot.create(:appeal, :hearing_docket, closest_regional_office: "RO39", veteran_file_number: veteran.file_number)
hearing_user = FactoryBot.create(:user, css_id: "HUOT_1", full_name: "Lendvai Huot", roles: ["Hearing Prep"])
hearing_day = FactoryBot.create(:hearing_day, judge: hearing_user, scheduled_for: 1.month.from_now, created_by: User.first, updated_by: User.first)
hearing = FactoryBot.create(:hearing, appeal: appeal, hearing_day: hearing_day, disposition: nil, adding_user: User.first, judge: hearing_user)
root_task = FactoryBot.create(:root_task, appeal: appeal)
hearing_task = FactoryBot.create(:hearing_task, parent: root_task, appeal: appeal, assigned_to: HearingsManagement.singleton, assigned_by: hearing_user)
association = FactoryBot.create(:hearing_task_association, hearing: hearing, hearing_task: hearing_task)
appeal.reload.treee
```

3. Confirm you cannot view the judge for a hearing, `localhost:3000/queue/appeals/${appeal.uuid}`
4. Switch to the user `TEAM_ADMIN at 101`.
5. Confirm you can still see the judge name in the hearings section.

### User Facing Changes
 - [ ] Screenshots of UI changes added to PR & Original Issue

View for a non-VSO user (order of judge and date has changed - I pinged design to ask if I should change it back) 
BEFORE
Master_NonVSOCanSeeJudge_MultipleHearings 
AFTER
NonVSOCanSeeJudge_MultipleHearings

View for a VSO user (it changed!)
BEFORE
Master_VSOCanSeeJudge_MultipleHearings

AFTER
VSOCantSeeJudge_MultipleHearings

### Code Documentation Updates
- [ ] Add or update code comments at the top of the class, module, and/or component.

### CODE REVIEW

File: app/serializers/hearings/appeal_hearing_serializer.rb [hunk 1]
```diff
@@ -6,8 +6,13 @@ class AppealHearingSerializer
   attribute :date, &:scheduled_for
   attribute :disposition
   attribute :external_id
-  attribute :held_by do |hearing|
-    hearing.judge.present? ? hearing.judge.full_name : ""
+  attribute :held_by do |hearing, params|
+    if params[:user]&.vso_employee?
```
[the remark this answers is not in the document]
Comment (n1zyy) [reply app/serializers/hearings/appeal_hearing_serializer.rb:10]: +1 to your judgment here. This is much easier to read.

That said, the old code was returning an empty string and now we return nil. Are they interchangeable as far as the UI is concerned?

I suppose we _could_ collapse the two conditionals, e.g.:
```ruby
params[:user]&.vso_employee? || !hearing.judge.present? ? nil : hearing.judge.full_name
```

I think that's still harder to read, though, so 👍 to what you have.

File: client/app/queue/utils.js [hunk 2]
```diff
@@ -304,7 +304,7 @@ export const prepareAppealIssuesForStore = (appeal) => {
 
 export const prepareAppealHearingsForStore = (appeal) =>
   appeal.attributes.hearings.map((hearing) => ({
-    heldBy: hearing.held_by,
+    heldBy: hearing.held_by && hearing.held_by,
```
Comment (n1zyy) [inline client/app/queue/utils.js:307]: This is done twice in this file, but I don't understand it. This looks like it's the same thing twice?

File: app/serializers/hearings/appeal_hearing_serializer.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (eileen-nava) [reply app/serializers/hearings/appeal_hearing_serializer.rb:10]: Good question!  With my current code, this is what displays when the hearing judge is not present.

Screen Shot 2021-08-20 at 4 28 24 PM

Here is what displays when I change the code to ret run an empty string for the scenario where the hearing judge is not present.
Screen Shot 2021-08-20 at 4 27 13 PM


In other words, there's no difference.

File: client/app/queue/utils.js (hunk 2 above)
Comment (eileen-nava) [reply client/app/queue/utils.js:307]: If “hearing.held_by” is a truthy value, `hearing.held_by` gets called.  If “hearing.held_by” is a falsey value (nil or false) then `hearing.held_by` does not get called.

I’m realizing the second scenario is equivalent to a situation where “hearing.held_by” is called and it returns “nil.”  Maybe I could just change these two snippets from `hearing.held_by && hearing.held_by` to `hearing.held_by`?  Thoughts?


I thought the linter would prefer `heldBy: hearing.held_by && hearing.held_by,` to `heldBy: hearing.held_by,` but I could be wrong.

Comment (eileen-nava) [reply client/app/queue/utils.js:307]: Oh, and empty strings are also falsey in JavaScript.  That was maybe more relevant before I changed the serializer. 🤔

**Review by pbradin — APPROVED** [review]
Looks really good Eileen!

**Pull request merged**